In [9]:
library(dplyr)
library(ggplot2)
library(parallel)
library(caret)

options(mc.cores=detectCores())

Loading required package: lattice


In [4]:
unzip("creditcardfraud.zip")

In [6]:
ccdat = read.csv('creditcard.csv')

Time V1         V2          V3        V4         V5          V6         
1 0    -1.3598071 -0.07278117 2.5363467  1.3781552 -0.33832077  0.46238778
2 0     1.1918571  0.26615071 0.1664801  0.4481541  0.06001765 -0.08236081
3 1    -1.3583541 -1.34016307 1.7732093  0.3797796 -0.50319813  1.80049938
4 1    -0.9662717 -0.18522601 1.7929933 -0.8632913 -0.01030888  1.24720317
5 2    -1.1582331  0.87773675 1.5487178  0.4030339 -0.40719338  0.09592146
6 2    -0.4259659  0.96052304 1.1411093 -0.1682521  0.42098688 -0.02972755
  V7          V8          V9         ⋯ V21          V22          V23        
1  0.23959855  0.09869790  0.3637870 ⋯ -0.018306778  0.277837576 -0.11047391
2 -0.07880298  0.08510165 -0.2554251 ⋯ -0.225775248 -0.638671953  0.10128802
3  0.79146096  0.24767579 -1.5146543 ⋯  0.247998153  0.771679402  0.90941226
4  0.23760894  0.37743587 -1.3870241 ⋯ -0.108300452  0.005273597 -0.19032052
5  0.59294075 -0.27053268  0.8177393 ⋯ -0.009430697  0.798278495 -0.13745808
6  0.47620095  0.26031433 -0.5686714 ⋯ -0.208253515 -0.559824796 -0.02639767
  V24         V25        V26        V27          V28         Amount Class
1  0.06692807  0.1285394 -0.1891148  0.133558377 -0.02105305 149.62 0    
2 -0.33984648  0.1671704  0.1258945 -0.008983099  0.01472417   2.69 0    
3 -0.68928096 -0.3276418 -0.1390966 -0.055352794 -0.05975184 378.66 0    
4 -1.17557533  0.6473760 -0.2219288  0.062722849  0.06145763 123.50 0    
5  0.14126698 -0.2060096  0.5022922  0.219422230  0.21515315  69.99 0    
6 -0.37142658 -0.2327938  0.1059148  0.253844225  0.08108026   3.67 0

In [7]:
ccdat %>% dim

[1] 284807     31

Looks like we have 284k rows and 31 columns.

In [8]:
cdat.cls = colnames(ccdat)

[1] "Time"   "V1"     "V2"     "V3"     "V4"     "V5"     "V6"     "V7"    
 [9] "V8"     "V9"     "V10"    "V11"    "V12"    "V13"    "V14"    "V15"   
[17] "V16"    "V17"    "V18"    "V19"    "V20"    "V21"    "V22"    "V23"   
[25] "V24"    "V25"    "V26"    "V27"    "V28"    "Amount" "Class"

Okay, let's check the column types. 

In [12]:
lapply(ccdat, class) %>% unique

[[1]]
[1] "numeric"

[[2]]
[1] "integer"

Looks like the 'Class' column is the only integer and everything else is numeric. 

Let's convert Class to a factor (discrete levels)

In [15]:
ccdat$Class = as.factor(ccdat$Class)

In [16]:
ccdat$Class %>% table

.
     0      1 
284315    492 

We have 492 of calss '1'. Safe to assume, these are the fraudulent transactions. 